kratos/modified_shape_functions/modified_shape_functions.cpp

```diff
@@ -0,0 +1,359 @@
+    };
+
+    // Computes all the shape function values for the negative side.
+    void ModifiedShapeFunctions::GetNegativeSideShapeFunctionsAndGradientsValues(
+        Matrix &rNegativeSideShapeFunctionsValues,
+        std::vector<Matrix> &rNegativeSideShapeFunctionsGradientsValues,
+        Vector &rNegativeSideWeightsValues,
+        const IntegrationMethodType IntegrationMethod) {
+
+        KRATOS_ERROR << "Calling the base class GetShapeFunctionsAndGradientsValuesNegativeSide method. Call the specific geometry one.";
+    };
+
+    // Computes all the shape function values for the positive interface side.
+    void ModifiedShapeFunctions::GetInterfacePositiveSideShapeFunctionsAndGradientsValues(
+        Matrix &rInterfacePositiveSideShapeFunctionsValues,
+        std::vector<Matrix> &rPositiveInterfaceSideShapeFunctionsGradientsValues,
+        Vector &rPositiveInterfaceSideWeightsValues,
+        const IntegrationMethodType IntegrationMethod) {
+
+        KRATOS_ERROR << "Calling the base class GetInterfaceShapeFunctionsAndGradientsValuesPositiveSide method. Call the specific geometry one.";
+    };
+
+    // Computes all the shape function values for the negative interface side.
+    void ModifiedShapeFunctions::GetInterfaceNegativeSideShapeFunctionsAndGradientsValues(
+        Matrix &rInterfaceNegativeSideShapeFunctionsValues,
+        std::vector<Matrix> &rInterfaceNegativeSideShapeFunctionsGradientsValues,
+        Vector &rInterfaceNegativeSideWeightsValues,
+        const IntegrationMethodType IntegrationMethod) {
+
+        KRATOS_ERROR << "Calling the base class GetInterfaceShapeFunctionsAndGradientsValuesNegativeSide method. Call the specific geometry one.";
+    };
+
+    // Computes the positive side interface outwards unit normal vector values.
+    void ModifiedShapeFunctions::GetPositiveSideInterfaceUnitNormals(
+        std::vector<Vector> &rPositiveSideInterfaceUnitNormal,
+        const IntegrationMethodType IntegrationMethod) {
+
+        KRATOS_ERROR << "Calling the base class GetPositiveSideInterfaceUnitNormals method. Call the specific geometry one.";
+    };
+
+    // Computes the positive side interface outwards unit normal vector values.
+    void ModifiedShapeFunctions::GetNegativeSideInterfaceUnitNormals(
+        std::vector<Vector> &rNegativeSideInterfaceUnitNormal,
+        const IntegrationMethodType IntegrationMethod) {
+
+        KRATOS_ERROR << "Calling the base class GetNegativeSideInterfaceUnitNormals method. Call the specific geometry one.";
+    };
+
+    // Returns true if the element is splitting
+    bool ModifiedShapeFunctions::IsSplit() {
+
+        KRATOS_ERROR << "Calling the base class IsSplit method. Call the specific geometry one.";
+    };
+
+    // Sets the condensation matrix to transform the subdivsion values to entire element ones.
+    void ModifiedShapeFunctions::SetIntersectionPointsCondensationMatrix(
+        Matrix& rIntPointCondMatrix,
+        const int EdgeNodeI[],
```

Comment:
i would use 

boost::numeric::ublas::vector<int>

here instead of passing c arrays. (C arrays are evillllll)